import-service/src/libs/s3-functions.ts

```diff
@@ -0,0 +1,65 @@
+import { S3Client, DeleteObjectCommand, CopyObjectCommand } from "@aws-sdk/client-s3";
+
+const csv = require('csv-parser');
+
+const parseFile = async (S3, params) => {
+  return new Promise<void>(async (resolve, reject) => {
+    const S3Stream = S3.getObject(params).createReadStream();
+    const { Bucket, Key } = params;
+
+    S3Stream
+      .pipe(csv())
+      .on('data', data => {
+        console.log(`${Bucket}: ${Key}`, data);
+      })
+      .on('error', error => {
+        console.log(`${Bucket}: ${Key} error: `, error);
+        reject(new Error(error));
+      })
+      .on('end', () => {
+        console.log(`${Bucket}: ${Key} parsing completed`);
+        resolve();
+      })
+  });
+}
+
+const copyParsedFile = async (S3, params) => {
+  return new Promise<void>(async (resolve, reject) => {
+    const { Bucket, Key } = params;
+
+    const copyParams = {
+      CopySource: `${Bucket}/${Key}`,
+      Bucket,
+      Key: Key.replace('uploaded', 'parsed')
+    }
+
+    try {
```

Comment:
Do not mix try/catch and resolve/reject, here as you return Promise try to use resolve() if successful, and reject if an error occurs.